Azure/azure-cli
# [Role] Add long-summary/warning for commands generating credentials

## Description

- `az ad sp create-for-rbac`
- `az ad sp credential reset`
- `az ad app credential reset`

These commands may 

- show password in the terminal, or 
- generate a certificate and save it to the local file system

The user may accidentally check the credentials into the source control.

```sh
> az ad sp create-for-rbac
Creating a role assignment under the scope of "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590"
{
  "appId": "d488e1e6-1622-46fd-949b-91c1b51f7d10",
  "displayName": "azure-cli-2020-11-17-08-17-11",
  "name": "http://azure-cli-2020-11-17-08-17-11",
  "password": "uLbYRPXc8~...",                        ## credential here
  "tenant": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a"
}

> az ad sp credential reset --name 8873edb1-8a68-48b6-b374-a598b4b0a21e
{
  "appId": "8873edb1-8a68-48b6-b374-a598b4b0a21e",
  "name": "8873edb1-8a68-48b6-b374-a598b4b0a21e",
  "password": "uLbYRPXc8~...",                        ## credential here
  "tenant": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a"
}

> az ad app credential reset --id f924487a-8bde-4fe1-96f5-c2124d082e51
{
  "appId": "f924487a-8bde-4fe1-96f5-c2124d082e51",
  "name": "f924487a-8bde-4fe1-96f5-c2124d082e51",
  "password": "uLbYRPXc8~...",                        ## credential here
  "tenant": "54826b22-38d6-4fb2-bad9-b7b93a3e9c5a"
}
```

This PRs explicitly tells the user about the behavior via 

### help
```
The output includes credentials that you must protect. Be sure that you do not include these
credentials in your code or check the credentials into your source control. As an
alternative, consider using [managed identities](https://aka.ms/azadsp-managed-identities)
if available to avoid the need to use credentials.
```

### warning

```
The output includes credentials that you must protect. Be sure that you do not include these 
credentials in your code or check the credentials into your source control. For more information, 
see https://aka.ms/azadsp-cli
```

## Testing Guide

```
az ad sp create-for-rbac -h
az ad sp create-for-rbac
az ad sp credential reset -h
az ad sp credential reset --name <appID>
az ad app credential reset -h
az ad app credential reset --id <appID>
```

ℹ Whether `az ad sp create-for-rbac` should assign **Contributor** role by default is still under debate. (#16081)
